python/graphstorm/gconstruct/transform.py

```diff
@@ -97,6 +101,74 @@ def collect_info(self, info):
             Information to be collected
         """
 
+class CategoricalTransform(TwoPhaseFeatTransform):
+    """ Convert the data into categorical values.
+
+    The categorical values are stored as integers.
+
+    Parameters
+    ----------
+    col_name : str
+        The name of the column.
+    feat_name : str
+        The name of the feature.
+    """
+    def __init__(self, col_name, feat_name):
+        self._val_dict = {}
+        super(CategoricalTransform, self).__init__(col_name, feat_name)
+
+    def pre_process(self, feats):
+        """ Pre-process data
+
+        Parameters
+        ----------
+        feats: np.array
+            Data to be processed
+        """
+        return {self.feat_name: np.unique(feats)}
+
+    def update_info(self, info):
+        """ Store global information for the second phase data processing
+
+        Parameters
+        ----------
+        info: list
+            Information to be collected
+        """
+        self._val_dict = {key: i for i, key in enumerate(np.unique(np.concatenate(info)))}
+
+    def __call__(self, feats):
+        """ Assign IDs to categorical values.
+
+        Parameters
+        ----------
+        feats : np array
+            Data with categorical values.
+
+        Returns
+        -------
+        np.array
+        """
+        return {self.feat_name: np.array([self._val_dict[val] for val in feats])}
```

Comment:
This code only handle the case when each node belongs to only one category.

Reply:
this is to handle categorical values. to handle multiple categorical values, we need to define how multiple categorical values are stored.